src/Http/Controllers/Admin/XlsformTemplateCrudController.php

```diff
@@ -158,6 +164,27 @@ protected function setupUpdateOperation(): void
             ->type('upload')
             ->upload(true)
         ->validationRules('nullable');
+
+
+        // TOOD: move below code segment to select() function
+
+        // get current entry
+        $entry = $this->crud->getCurrentEntry();
+
+        // read excel files into an array of excel sheets
+        $sheets = Excel::toArray(new XlsformSurveyImport, $entry->xlsfile);
+
+        // get the first excel data sheet, as it is always the survey template
+        $sheet1 = $sheets[0];
```

Comment:
One possible issue here is that the survey worksheet is not always the first sheet. (It is if you're following the standard templates, but people often don't do that...), so we ideally should check the worksheet name. Perhaps you can use the `XlsformImport` here instead of the `XlsformSurvey` import, as that's already setup for multiple sheets and specifically looks for the 'survey' and 'choices' worksheet.

Reply:
Thank you. I have modified it as per your advice, it is now referring to the 'survey' worksheet.